pypeit/spec2dobj.py

```diff
@@ -103,38 +104,42 @@ class Spec2DObj(datamodel.DataContainer):
                  'detector': dict(otype=(DetectorContainer, Mosaic),
                                   descr='Detector or Mosaic metadata') }
 
+    internals = ['calibs',              # Dictionary containing the processed calibration frames
+                 'process_steps',       # List of image processing steps
+                 'head0'                # Raw header
+                ]
+
     @classmethod
-    def from_file(cls, file, detname, chk_version=True):
+    def from_file(cls, file, detname, **kwargs):
```

Comment:
yeah to `kwargs`

Reply:
Is the set of parameters here really so large that we need kwargs. I really don't like kwargs. It just makes for code that is difficult to read and can lead to errors since name collisions between kwarg params and function params don't get caught.

Reply:
In this case, I agree.  Changed.